google/cloud/storage/internal/signed_url_requests.h

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_SIGNED_URL_REQUESTS_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_SIGNED_URL_REQUESTS_H_
+
+#include "google/cloud/storage/hashing_options.h"
+#include "google/cloud/storage/internal/generic_request.h"
+#include "google/cloud/storage/signed_url_options.h"
+#include "google/cloud/storage/well_known_parameters.h"
+#include <iosfwd>
+#include <map>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+/**
+ * Requests the Google Cloud Storage service account for a project.
+ */
+class SignUrlRequest {
+ public:
+  SignUrlRequest() = default;
+  explicit SignUrlRequest(std::string verb, std::string bucket_name,
+                          std::string object_name)
+      : verb_(std::move(verb)),
+        bucket_name_(std::move(bucket_name)),
+        object_name_(std::move(object_name)) {}
+
+  std::string const& verb() const { return verb_; }
+  std::string const& bucket_name() const { return bucket_name_; }
+  std::string const& object_name() const { return object_name_; }
+
+  /// Creates the blob to be signed.
+  std::string CreateBlob() const;
```

Comment:
`CreateBlob()` sounds like something that would upload content or create a file.  Can we name this something like `StringToSign()`?

Reply:
Done.